src/solve.jl

```diff
@@ -68,6 +68,14 @@ end
 ```
 """
 function solve(prob::OptimizationProblem, alg, args...; kwargs...)
+    !isbounded(alg) && (!isnothing(prob.lb) || !isnothing(prob.ub)) &&
+        throw(ArgumentError("The algorithm $(typeof(alg)) does not support box constraints."))
+    !isconstrained(alg) && !isnothing(prob.f.cons) &&
+        throw(ArgumentError("The algorithm $(typeof(alg)) does not support constraints."))
+    isconstrained(alg) && isnothing(prob.f.cons) &&
+        throw(ArgumentError("The algorithm $(typeof(alg)) requires constraints, pass them with the `cons` kwarg in `OptimizationFunction`."))
+    !callbacks_support(alg) && !isnothing(kwargs[:callback]) &&
+        throw(ArgumentError("The algorithm $(typeof(alg)) does not support callbacks."))
```

Comment:
can these get custom exceptions like the other throws, and more informative like our other error messages?